subsystem/vis_telemetry/run.py

```diff
@@ -0,0 +1,85 @@
+import os
+import math
+import datetime
+from common import invoke_main, sort_data, idemp_mkdir, write_status, process_gpu_telemetry, process_cpu_telemetry
+from dashboard_info import DashboardInfo
+from plot_util import PlotBuilder, UnitType, PlotType
+
+
+def extract_float(s:str) -> float:
+    s = ''.join(list(filter(lambda x: x.isdigit() or x in ('+', '-', '.', 'e'), s)))
+    try:
+        return float(s)
+    except:
```

Comment:
Some data returned by the sensors is `N/A` (or other non-numerical data like `ALARM`), which cannot be parsed to float. This is for filtering out those data that cannot be used to plot a graph.

Reply:
Can we try to list these exceptions and pick them out explicitly? I would be worried about having silent failures if we change anything later

Reply:
Sure. Should we still return `nan` for those entries while dumping the exceptions to files?

Reply:
Sure. Should we dump the exception message to a file (like `status.json`) and return `nan`?

Reply:
Not exactly; what I mean is that the functions responsible for parsing commands should have an explicit list of non-numerical values that can come up and recognize them without raising an exception (so that way if the parser encounters a value we have not thought about before, there would be an uncaught exception and we would have to debug it)

Reply:
Sure. But these fields vary according to machines. So we may want to list those fields explicitly in the configuration of the subsystem.

Reply:
I thought about that, and one more problem I'm considering is maybe it is not always the case that a query of a certain data field only returns non-numerical values, so we may encounter a non-numerical data after several queries that return numerical value.

Reply:
You should try to explicitly check for those non-numerical values instead of relying on the exceptions (again, my worry is having a mistake get silently caught by the catch)

Reply:
I got your point. I'll make the changes.